grammar.js

```diff
@@ -511,6 +546,7 @@ module.exports = grammar({
     do_loop_statement: $ => seq(
       optional($.block_label_start_expression),
       caseInsensitive('do'),
+      optional($.while_statement),
       optional($.loop_control_expression),
```

Comment:
We might want this to be an `optional(choice(..))` instead of two optionals. I think it could allow for some odd invalid syntax.

Reply:
Agreed, I have made the change.